src/tests/nativeaot/SmokeTests/Preinitialization/Preinitialization.cs

```diff
@@ -404,7 +404,15 @@ public ReferenceType(int intValue, double doubleValue)
 
     public static void Run()
     {
-        Assert.IsPreinitialized(typeof(TestReferenceTypeAllocation));
+        if (RuntimeInformation.ProcessArchitecture is Architecture.Arm or Architecture.Wasm)
+        {
+            // Because of the double field, this is not preinitialized
+            Assert.IsLazyInitialized(typeof(TestReferenceTypeAllocation));
+        }
+        else
+        {
+            Assert.IsPreinitialized(typeof(TestReferenceTypeAllocation));
+        }
```

Comment:
Could you add a test for `long/ulong` too?

Reply:
This is enough to exercise we go through the align8 paths. Testing the details of how align8 works is done elsewhere.